synapsegraph.py

```diff
@@ -0,0 +1,224 @@
+import threading
+import argparse
+import logging
+import json
+import sys
+from collections import OrderedDict
+import multiprocessing
+import UserDict
+import uuid
+
+import synapseclient
+import synapseutils
+
+import GraphToNeo4j
+
+syn = synapseclient.login(silent=True)
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+IGNOREME_NODETYPES = ['org.sagebionetworks.repo.model.Project',
+                      'org.sagebionetworks.repo.model.Preview']
+
+SKIP_LIST = ['syn582072', 'syn3218329', 'syn2044761', 'syn2351328', 'syn1450028']
+
+class MyEntity(UserDict.IterableUserDict):
+    """A dictionary representation of a Synapse entity.
+
+    Annotations are preprocessed for those that are containers and returned the first item.
+
+    """
+
+    def __init__(self, syn, d, projectId=None, benefactorId=None):
+        self._syn = syn
+
+        UserDict.IterableUserDict.__init__(self, d)
+
+        for key in self.data.keys():
+            if type(self.data[key]) is list and len(self.data[key]) > 0:
+                self.data[key] = self.data[key][0]
+
+        self.data.pop('annotations')
+
+        # If not supplied, get the project and benefactor of the entity
+        self.data['projectId'] = projectId or self._getProjectId(self.data['id'])
+        self.data['benefactorId'] = benefactorId or self._getBenefactorId(self.data['id'])
```

Comment:
this doesn't change the issue of user passing in bad project ID, so leaving as is.